src/providers/ogr/qgsogrprovider.cpp

```diff
@@ -4348,6 +4393,29 @@ void QgsOgrProviderUtils::invalidateCachedDatasets( const QString &dsName )
   }
 }
 
+
+QgsOgrDatasetSharedPtr QgsOgrProviderUtils::getAlreadyOpenedDataset( const QString &dsName )
+{
+  QMutexLocker locker( &sGlobalMutex );
+  for ( auto iter = sMapSharedDS.begin(); iter != sMapSharedDS.end(); ++iter )
+  {
+    auto ident = iter.key();
+    if ( ident.dsName == dsName && ident.updateMode )
+    {
+      // Browse through this list, to look for the first available DatasetWithLayers*
+      // instance that is in update mode (hoping there's only one...)
+      auto &datasetList = iter.value();
+      Q_FOREACH ( QgsOgrProviderUtils::DatasetWithLayers *ds, datasetList )
```

Comment:
Can this be a `const auto &` and a range based for loop?

Reply:
fixed